src/components/TextField/TextField.tsx

```diff
@@ -406,6 +409,15 @@ export function TextField({
     monospaced && styles.monospaced,
   );
 
+  const handleOnFocus = () => {
+    if (selectAllTextOnFocus) {
+      inputRef?.current?.select();
```

Comment:
Question about the use of optional chaining here - would this handler need them if we know `inputRef` is defined, and `current` is also defined when onFocus is called? 

I could be wrong! :)

Reply:
Great question! If you look at where inputRef is defined: 

```typescript
  const inputRef = useRef<HTMLInputElement | HTMLTextAreaElement>(null);

```

It passes `null` as the default value because `useRef` can only be null or in this case an Input or TextArea element! This means that `inputRef` can be null so Typescript catches that for us and yells! Hehe. So the optional chaining is needed here 🥳

Reply:
Refs are a little tricky, the default value is `null`, but the default value is the `current` value. The type for `useRef` would look something like this

```ts
type UseRef<T> = (val: T) => {current: T};
```

Because of this, `inputRef` is always defined. But the current at `inputRef.current` defaults to null 😄 

```suggestion
      inputRef.current?.select();
```

Reply:
OOOH!!! Thank-you for this!

Reply:
lttp ;p 

Looks good! Thanks!